components/script/webdriver_handlers.rs

```diff
@@ -81,8 +84,42 @@ pub unsafe fn jsval_to_webdriver(cx: *mut JSContext, val: HandleValue) -> WebDri
                 _ => unreachable!(),
             };
         Ok(WebDriverJSValue::String(String::from(string)))
-    } else if val.get().is_null() {
-        Ok(WebDriverJSValue::Null)
+    } else if val.get().is_object() {
+        let object: *mut JSObject = match FromJSValConvertible::from_jsval(cx, val, ()).unwrap() {
```

Comment:
Let's use `rooted!(in(cx) let object = ...);` here, and then we can use `object.handle()` later instead.